com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs

```diff
@@ -965,5 +957,94 @@ internal static void SortStable(List<ScriptableRenderPass> list)
                 list[j + 1] = curr;
             }
         }
+
+        internal struct RenderBlocks : IDisposable
+        {
+            private NativeArray<RenderPassEvent> m_BlockEventLimits;
+            private NativeArray<int> m_BlockRanges;
+            private NativeArray<int> m_BlockRangeLengths;
+            public RenderBlocks(List<ScriptableRenderPass> activeRenderPassQueue)
+            {
+                // Upper limits for each block. Each block will contains render passes with events below the limit.
+                m_BlockEventLimits = new NativeArray<RenderPassEvent>(k_RenderPassBlockCount, Allocator.Temp);
+                m_BlockRanges = new NativeArray<int>(m_BlockEventLimits.Length + 1, Allocator.Temp);
+                m_BlockRangeLengths = new NativeArray<int>(m_BlockRanges.Length, Allocator.Temp);
+
+                m_BlockEventLimits[RenderPassBlock.BeforeRendering] = RenderPassEvent.BeforeRenderingPrepasses;
+                m_BlockEventLimits[RenderPassBlock.MainRenderingOpaque] = RenderPassEvent.AfterRenderingOpaques;
+                m_BlockEventLimits[RenderPassBlock.MainRenderingTransparent] = RenderPassEvent.AfterRenderingPostProcessing;
+                m_BlockEventLimits[RenderPassBlock.AfterRendering] = (RenderPassEvent) Int32.MaxValue;
+
+                // blockRanges[0] is always 0
+                // blockRanges[i] is the index of the first RenderPass found in m_ActiveRenderPassQueue that has a ScriptableRenderPass.renderPassEvent higher than blockEventLimits[i] (i.e, should be executed after blockEventLimits[i])
+                // blockRanges[blockEventLimits.Length] is m_ActiveRenderPassQueue.Count
+                FillBlockRanges(activeRenderPassQueue);
+                m_BlockEventLimits.Dispose();
+
+                for (int i = 0; i < m_BlockRanges.Length - 1; i++)
+                {
+                    m_BlockRangeLengths[i] = m_BlockRanges[i + 1] - m_BlockRanges[i];
+                }
+            }
+
+            /// <summary>
+            ///  Dispose pattern implementation
+            /// </summary>
+            public void Dispose()
```

Comment:
I don't see where you are disposing these.

Reply:
On line 419: `using var renderBlocks = new RenderBlocks(m_ActiveRenderPassQueue);`
It's the `using` which implements the RAII like pattern here.
I did check it gets called.
Anyway this way you never forget to call a dispose explicitly at the end of the function.